protocols/v2/subprotocols/mining/src/new_mining_job.rs

```diff
@@ -139,3 +139,96 @@ impl<'d> GetSize for NewMiningJob<'d> {
             + self.merkle_root.get_size()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::tests::from_arbitrary_vec_to_array;
+    use binary_sv2::GetSize;
+    use core::convert::TryFrom;
+    use quickcheck_macros;
+
+    #[quickcheck_macros::quickcheck]
+    fn test_new_extended_mining_job(
+        channel_id: u32,
+        job_id: u32,
+        future_job: bool,
+        version: u32,
+        version_rolling_allowed: bool,
+        merkle_path: Vec<u8>,
+        mut coinbase_tx_prefix: Vec<u8>,
+        mut coinbase_tx_suffix: Vec<u8>,
+    ) -> bool {
+        let merkle_path = helpers::scan_to_u256_sequence(&merkle_path);
+        let coinbase_tx_prefix = helpers::bytes_to_b064k(&mut coinbase_tx_prefix);
+        let coinbase_tx_suffix = helpers::bytes_to_b064k(&mut coinbase_tx_suffix);
+        let nemj = NewExtendedMiningJob {
+            channel_id,
+            job_id,
+            future_job,
+            version,
+            version_rolling_allowed,
+            merkle_path: merkle_path.clone(),
+            coinbase_tx_prefix: coinbase_tx_prefix.clone(),
+            coinbase_tx_suffix: coinbase_tx_suffix.clone(),
+        };
+        let static_nmj = nemj.as_static();
+        static_nmj.channel_id == nemj.channel_id
+            && static_nmj.job_id == nemj.job_id
+            && static_nmj.future_job == nemj.future_job
+            && static_nmj.version == nemj.version
+            && static_nmj.version_rolling_allowed == nemj.version_rolling_allowed
+            && static_nmj.merkle_path == merkle_path
+            && static_nmj.coinbase_tx_prefix == coinbase_tx_prefix
+            && static_nmj.coinbase_tx_suffix == coinbase_tx_suffix
+    }
+
+    #[quickcheck_macros::quickcheck]
+    fn test_new_mining_job(
+        channel_id: u32,
+        job_id: u32,
+        future_job: bool,
+        version: u32,
+        merkle_root: Vec<u8>,
+    ) -> bool {
+        let merkle_root = from_arbitrary_vec_to_array(merkle_root);
+        let nmj = NewMiningJob {
+            channel_id,
+            job_id,
+            future_job,
+            version,
+            merkle_root: B032::try_from(merkle_root.to_vec())
+                .expect("NewMiningJob: failed to convert merkle_root to B032"),
+        };
+        let static_nmj = nmj.clone().as_static();
+        static_nmj.channel_id == nmj.channel_id
+            && static_nmj.job_id == nmj.job_id
+            && static_nmj.future_job == nmj.future_job
+            && static_nmj.version == nmj.version
+            && static_nmj.merkle_root == nmj.merkle_root
+    }
+
+    #[test]
+    fn test() {
+        "placeholder";
```

Comment:
What does this do?

Reply:
I saw the text for the placeholder below - can you use the same explanation here?

Reply:
will remove this. I had to add this for rust-analyzer to realize the test module for quick testing since it doesnt recognize the quickcheck attr macro